### DISCUSSION TITLE
Scalability

### DISCUSSION THREAD

**Discussion by selarem** [idaholab/moose#23187] [Q&A Tools] [answered]
Hi all,
I would like to know if anyone has a scalability curve of moose.
I will ask for computing time on a national computing center and would like to give them 
an approximation of our needs.
Our simulations couple Crystal plasticity (1000 grains) with phase field damage and CZM. 
The number of DOF goes from 2 millions to 20 millions.
thanks for any help
regards,
sab

**Comment by GiudGiud** [comment, marked as answer]
Hello

Scalability is application dependent as the numerical system is not solved the same way for every physics. 
@permcody might know the right paper for scalability of the phase field module 
The rule of thumb we use is 20k dofs *at least* per processor, so for 20M we would say use less than 1,000 processors but it's likely to work with much less. 

Guillaume

**Reply by hugary1995** [reply]
See the moose softwarex paper. The framework scales very, very well to tens of thousands cores. However, I have limited success in scaling a crystalplasiticity + czm simulation. The bottleneck is a scalable linear solver. I think you might go to several hundreds cores, beyond that is gonna be a hard stretch. There's something else that didn't scale well when I have both czm (mesh broken) and periodic strain conditions. I can't recall exactly what that was on top of my head right now.
